### ISSUE TITLE
Control Tridactyl via CLI through native messenger

### ISSUE THREAD

**Issue by bovine3dom** [cmcaine/tridactyl#780]
E.g. `tridactyl "winopen -private"`.

Would fix #775.

**Comment by chrispickard** [comment]
I would be interested in looking at this, we would have to convert the native part of the extension into a connection based extension, I think.

does this sound right?

I think that would be the first step.

Then we could allow communication from the CLI

**Comment by bovine3dom** [comment]
Yeah. I think we've talked about it before somewhere - try searching for sockets or daemon in issues or in Riot chat.

**Comment by bovine3dom** [comment]
I briefly thought "hey we can add a protocol handler to Tridactyl" to fix this via - `ext+tridactyl://` https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/protocol_handlers#content

then `firefox "ext+tridactyl://[whatever]"` would work

but that's a terrible idea as any site could redirect to a link with that protocol and run whatever they want on the user's machine.

I just wanted to document what a bad idea it was to stop me from thinking of it again and implementing it

**Comment by gerardbosch** [comment]
I've just found this issue and I'm wondering if it could be related in any way to this question: https://github.com/tridactyl/tridactyl/discussions/4012
